src/stream/src/executor/managed_state/aggregation/string_agg.rs

```diff
@@ -12,610 +12,530 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#![allow(dead_code)]
-
-use std::collections::BTreeMap;
-
-use bytes::Bytes;
-use itertools::Itertools;
-use risingwave_common::array::stream_chunk::{Op, Ops};
-use risingwave_common::array::ArrayImpl;
+use std::collections::{BTreeSet, HashMap};
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::pin_mut;
+use futures_async_stream::for_await;
+use risingwave_common::array::stream_chunk::Ops;
+use risingwave_common::array::Op::{Delete, Insert, UpdateDelete, UpdateInsert};
+use risingwave_common::array::{ArrayImpl, Row};
 use risingwave_common::buffer::Bitmap;
-use risingwave_common::types::{DataType, Datum, ScalarImpl};
-use risingwave_common::util::ordered::OrderedArraysSerializer;
-use risingwave_common::util::value_encoding::{deserialize_cell, serialize_cell};
-use risingwave_storage::storage_value::StorageValue;
+use risingwave_common::types::{Datum, ScalarImpl};
+use risingwave_common::util::sort_util::{DescOrderedRow, OrderPair, OrderType};
 use risingwave_storage::table::state_table::RowBasedStateTable;
-use risingwave_storage::write_batch::WriteBatch;
-use risingwave_storage::{Keyspace, StateStore};
+use risingwave_storage::StateStore;
+
+use super::ManagedTableState;
+use crate::executor::aggregation::AggCall;
+use crate::executor::error::StreamExecutorResult;
+use crate::executor::PkIndices;
+
+#[derive(Debug)]
+struct Cache {
+    synced: bool, // `false` means not synced from state table (cold start)
+    order_pairs: Arc<Vec<OrderPair>>,
```

Comment:
Can we provide some comments for why `order_pairs` is introduced and how it used? A little unclear.

Reply:
Comments added, hope it's clear now.